lib/src/main.dart

```diff
@@ -218,3 +219,26 @@ class MainPage extends StatelessWidget {
     );
   }
 }
+
+/// When thumbnail is available draw image using full width available.
+/// 
+/// TODO: replace thumbnail with hi-res image.
+Widget addImage(Submission submission) {
+  return submission.thumbnail.toString() != 'self'

```

Comment:
Can you move this check outside of this function? Just inline it on line 140.
Use `submission.isSelf` instead.